core/platform/email/gae_email_services_test.py

```diff
@@ -36,10 +45,201 @@ def test_sending_email(self):
             to=feconf.ADMIN_EMAIL_ADDRESS)
         self.assertEqual(0, len(messages))
 
+        messages = self.mail_stub.get_sent_messages(
+            to=feconf.ADMIN_EMAIL_ADDRESS)
+        self.assertEqual(0, len(messages))
         with self.swap(feconf, 'CAN_SEND_EMAILS', True):
             gae_email_services.send_mail(
-                feconf.SYSTEM_EMAIL_ADDRESS, feconf.ADMIN_EMAIL_ADDRESS,
+                feconf.SYSTEM_EMAIL_ADDRESS,
+                feconf.ADMIN_EMAIL_ADDRESS,
                 'subject', 'body', 'html', bcc_admin=False)
         messages = self.mail_stub.get_sent_messages(
             to=feconf.ADMIN_EMAIL_ADDRESS)
         self.assertEqual(1, len(messages))
+
+    def test_email_bcc_is_sent_if_bcc_admin_is_true(self):
+        """Tests that the email has ADMIN_EMAIL_ADDRESS as bcc
+        if bcc_admin is True.
+        """
+        messages = self.mail_stub.get_sent_messages(to=self.RECIPIENT_EMAIL)
+        self.assertEqual(0, len(messages))
+        with self.swap(feconf, 'CAN_SEND_EMAILS', True):
+            gae_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, self.RECIPIENT_EMAIL,
+                'subject', 'body', 'html', bcc_admin=True)
+        messages = self.mail_stub.get_sent_messages(to=self.RECIPIENT_EMAIL)
+        self.assertEqual(1, len(messages))
+        self.assertEqual(messages[0].bcc, feconf.ADMIN_EMAIL_ADDRESS)
+
+    def test_email_bcc_not_sent_if_bcc_admin_is_false(self):
+        # Tests that email has no bcc attribute if bcc_admin is False.
+        messages = self.mail_stub.get_sent_messages(to=self.RECIPIENT_EMAIL)
+        self.assertEqual(0, len(messages))
+        with self.swap(feconf, 'CAN_SEND_EMAILS', True):
+            gae_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, self.RECIPIENT_EMAIL,
+                'subject', 'body', 'html', bcc_admin=False)
+        messages = self.mail_stub.get_sent_messages(to=self.RECIPIENT_EMAIL)
+        self.assertEqual(1, len(messages))
+        self.assertFalse(hasattr(messages[0], 'bcc'))
+
+    def test_sending_email_with_reply_to_id_adds_reply_to_email(self):
+        # Tests that email reply_to if passed.
```

Comment:
Remove this line or move this to the place where you are passing the reply_to_id.

Reply:
Done.
Thanks!